packages/mui-base/src/Portal/Portal.js

```diff
@@ -48,7 +48,11 @@ const Portal = React.forwardRef(function Portal(props, ref) {
     return children;
   }
 
-  return mountNode ? ReactDOM.createPortal(children, mountNode) : mountNode;
+  return (
+    <React.Fragment>
```

Comment:
I have doubled check, this is not shown in the React dev tools, so no DX downside.